optuna/integration/tensorboard.py

```diff
@@ -74,9 +81,24 @@ def _add_distributions(
                     hp.IntInterval(param_distribution.low, param_distribution.high),
                 )
             elif isinstance(param_distribution, optuna.distributions.CategoricalDistribution):
+                choices = param_distribution.choices
+                dtype = type(choices[0])
+                if any(not isinstance(choice, dtype) for choice in choices):
+                    _logger.warning(
+                        "Choices contains mixed types, which is not supported by TensorBoard."
+                        "Converting all choices to strings."
+                    )
+                    choices = tuple(map(str, choices))
+                elif dtype not in (int, float, bool, str):
+                    _logger.warning(
+                        f"Choices are of type {dtype}, which is not supported by TensorBoard."
+                        "Converting all choices to strings."
+                    )
+                    choices = tuple(map(str, choices))
```

Comment:
Maybe I missed something, but after this conversion, all the element of `choices` will be str. So, I guess `dtype` should be updated to `str` as well.